netlogo-gui/src/main/agent/Turtle.java

```diff
@@ -1238,4 +1241,38 @@ public int alpha() {
   }
 
 
+  private List<Link> links = new ArrayList<>(0);
+
+  public void addLink(Link link) {
+    links.add(link);
+  }
+
+  public boolean removeLink(Link link) {
+    return links.remove(link);
+  }
+
+  public Link[] links() {
+    return links.toArray(new Link[links.size()]);
+  }
+
+  public Link[] selectLinks(boolean undirected, boolean out, boolean in, AgentSet breed) {
+    Link[] result = new Link[links.size()];
+    int writeTo = 0;
+    for (Link link : links) {
+      boolean isDir = link.isDirectedLink();
+      boolean isOut = isDir && link.end1 == this;
+      boolean isIn = isDir && link.end2 == this;
+      boolean passesDirectedness = (undirected && !isDir) || (out && isOut) || (in && isIn);
+      boolean passesBreededness = breed == world.links() || breed == link.getBreed();
+      if (passesBreededness && passesDirectedness) {
```

Comment:
Are we expecting this code to be in a tight loop? (Either the `for` loop here, or something calling `selectLinks` itself in a tight loop).